### PR TITLE
Setup Angular Storybook

### LINKED ISSUE #1833: Set up Storybook - Angular NDS
**User Story:**

**Notes:**
Risk: Plug in that might not work in NG

**AC:**
- Get storybook running
- Angular SB config file, needs to live in a new NG directory
- import storybook theme
- set up storybook plugins

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Sets up the Angular Storybook environment and updates the Spark angular file system.

### Associated Issue 
Fixes #1833 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [ ] Update Spark Docs Angular

### Code
 - [x] Build Component in Spark Angular
- [x] Build Spark Angular Storybook

### Accessibility
- nothing to add here for now

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)
  
### Manual Testing Plan
- nothing to add here for now

### Design Review
- nothing to add here for now

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -11,8 +11,7 @@
     "url": "https://github.com/sparkdesignsystem/spark-design-system"
   },
   "dependencies": {
-    "@sparkdesignsystem/spark": "^11.0.0",
-    "@sparkdesignsystem/spark-react": "^1.0.1",
+    "@sparkdesignsystem/spark": "^12.0.0",
```
Comment (RCopeland) [inline react/package.json]: shouldnt have this as a dep, since we're bringing in local files

Comment (afebbraro) [reply react/package.json]: I was thinking for the cases where the local files bring it in as a dep. What do you think?

File: vanilla/package.json [hunk 2]
```diff
@@ -10,10 +10,11 @@
     "type": "git",
     "url": "https://github.com/sparkdesignsystem/spark-design-system"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@sparkdesignsystem/spark": "^12.1.1"
```
Comment (RCopeland) [inline vanilla/package.json]: we dont want to load the dep here either, since we're loading src files

File: angular/.storybook/config.js [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
-import '!style-loader!css-loader!sass-loader!@sparkdesignsystem/spark/_spark.scss';
-import { configure, addParameters } from '@storybook/angular';
-import { themes } from '@storybook/theming';
-import sparkTheme from "../storybook-spark-theme";
+import { configure, addDecorator, addParameters } from '@storybook/angular';
+import { withA11y } from '@storybook/addon-a11y';
+import sparkTheme from "../../storybook-spark-theme";
 
+addDecorator(withA11y);
+// addDecorator(story => <div className="sprk-o-Box">{story()}</div>)
```
Comment (RCopeland) [inline angular/.storybook/config.js]: should remove the comment

File: angular/.storybook/presets.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module.exports = [
+  '@storybook/addon-docs/angular/preset',
+  '@storybook/preset-scss'
```
Comment (RCopeland) [inline angular/.storybook/presets.js]: should do it manually so that postcss-loader can get used

File: angular/package.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  "version": "1.0.0",
+  "bugs": {
+    "url": "https://github.com/sparkdesignsystem/spark-design-system/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/sparkdesignsystem/spark-design-system"
+  },
+  "dependencies": {
+    "@sparkdesignsystem/spark": "^12.1.1",
```
Comment (RCopeland) [inline angular/package.json]: this dep should get removed

File: angular/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "target": "es5",
+    "typeRoots": [
+      "node_modules/@types"
+    ],
+    "lib": [
+      "es2017",
+      "dom"
+    ],
+    "paths": {
+      "@sparkdesignsystem/*": [
```
Comment (RCopeland) [inline angular/tsconfig.json]: wont need this path after the dep is removed

File: spark/manifests/spark-angular/spark-angular.module.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
+import { environment } from './src/environment/environment';
+window['sprk'] = { version: { angular: environment.version } } || {};
```
Comment (RCopeland) [inline spark/manifests/spark-angular/spark-angular.module.ts]: probably have to wrap in a window check

File: angular/.storybook/config.js (hunk 3 above)
Comment (afebbraro) [reply angular/.storybook/config.js]: added a todo

File: angular/.storybook/presets.js (hunk 4 above)
Comment (afebbraro) [reply angular/.storybook/presets.js]: The angular SB doesnt allow full-control mode so we'll have to work on this after our thing.

**Pull request merged**